### PR TITLE
 feat: add a `datetimerangequickpicker`  ,support timezone,clearable,quickchange

### PR DESCRIPTION
**Pull request by xyfy**
style: set `datetimerangepicker` class

feat: add a new  `datetimerangequickpicker`  ,support timezone,clearable,quickchange

### CODE REVIEW

File: src/Masa.Stack.Components/MasaComponentBase.cs [hunk 1]
```diff
@@ -44,6 +45,11 @@ protected string DT(string key)
         return TranslateProvider.DT(key);
     }
 
+    protected string DateTimeRange(string key)
+    {
+        return I18n.T("DateTimeRange", key);
+    }
+
```
Comment (capdiem) [inline src/Masa.Stack.Components/MasaComponentBase.cs:52]: 方法不对，完全看不出来是给I18N用的

File: src/Masa.Stack.Components/Shared/IntegrationComponents/DateTime/SDateTimeRangePicker.razor [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
-                </div>
-            </ActivatorContent>
-            <ChildContent>
-                <SDateTimePicker
-                    @bind-Value=InternalEndTime
-                    OutputTimezoneOffset ="ValueTimezoneOffset"
-                    DisplayTimezoneOffset="DisplayTimezoneOffset"
-                    Min="StartTime">
-                    <MButton Text Color="primary" OnClick="async() => await UpdateEndTimeAsync()">@T("Confirm")</MButton>
+
```
Comment (capdiem) [inline src/Masa.Stack.Components/Shared/IntegrationComponents/DateTime/SDateTimeRangePicker.razor:4]:
```suggestion
```

**Review by capdiem — CHANGES_REQUESTED** [review]
TL;DR

File: src/Masa.Stack.Components/Shared/IntegrationComponents/DateTime/SDateTimeRangeToolbar.razor [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    }
+
+    private async Task OnRelatedTimeSpanSelected(QuickRange item)
+    {
+        _quickRange = item;
+        await UpdateUseAbsoluteTimeState(false);
+        await UpdateRangeDateTimeToLatest();
+        if (DefaultQuickRangeChanged.HasDelegate)
+            await DefaultQuickRangeChanged.InvokeAsync(item.Key);
+        StateHasChanged();
```
Comment (capdiem) [inline src/Masa.Stack.Components/Shared/IntegrationComponents/DateTime/SDateTimeRangeToolbar.razor:259]: 会默认调StateHasChanged的
```suggestion
```

**Review by capdiem — CHANGES_REQUESTED** [review]

**Review by capdiem — APPROVED** [review]
